### DISCUSSION TITLE
Does Helium support Network Time Request?

### DISCUSSION THREAD

**Discussion by matthias-bs** [mcci-catena/arduino-lorawan#206] [Q&A]
With MCCI_LoRaWAN_LMIC_library/src/lmic/lmic.c/.h I can get the time from The Things Network, but not from Helium.
Is this functionality missing in Helium or is there any difference in using it?

**Comment by terrillmoore** [comment]
My guess is that it's not supported. Helium network capabilities are roughly equivalent to TTNv2, which also didn't support it.

**Reply by matthias-bs** [reply]
According to https://docs.helium.com/use-the-network/devices/: _The Helium network supports any LoRaWAN capable device meeting the v1.0.2, v1.0.3, or v1.0.4 specification._

According to a Dragino device user manual (however -  I do not consider this as an authoritative reference), a LoRaWAN server must support the v1.0.3 protocol to reply to LoRaWAN MAC Command DeviceTimeReq.

Hmmm....

**Reply by terrillmoore** [reply]
Sad to say, there is an ambiguity. "Supporting any device meeting 1.0.3", etc., just means that the NS will provide all mandatory features. According to the LoRaWAN spec, the network is not required to respond to `DeviceTimeReq`. 
 
It is possible to synch time using downlinks from your app.  Search for @cyberman54 in [arduino-lmic](https://github.com/mcci-catena/arduino-lmic), who reported doing just this using The Things Network (before TTN implemented `DeviceTimeReq`).

**Comment by matthias-bs** [comment]
Thank you for the quick response!

According to the Helium Console (Events->Uplink->MAC Commands), the DEVICE_TIME_REQ is correctly received.

![Helium Console: Events->Uplink->MAC Commands: DEVICE_TIME_REQ](https://user-images.githubusercontent.com/83612361/194726544-785c66e6-e04c-4e4f-a3ea-f3ef67117d97.png)

**Comment by terrillmoore** [comment]
Yep. But if the core network server doesn’t support it, you’re sort of out of luck.

 

--Terry
